libs/partners/ai21/tests/unit_tests/test_semantic_text_splitter.py

```diff
@@ -0,0 +1,130 @@
+    ids=[
+        "when_chunk_size_is_zero",
+        "when_chunk_size_is_large",
+        "when_chunk_size_is_small",
+    ],
+    argnames=["chunk_size", "expected_segmentation_len"],
+    argvalues=[
+        (0, 2),
+        (1000, 1),
+        (10, 2),
+    ],
+)
+def test_split_text__on_chunk_size(
+    chunk_size: int,
+    expected_segmentation_len: int,
+    mock_client_with_semantic_text_splitter: Mock,
+) -> None:
+    sts = AI21SemanticTextSplitter(
+        chunk_size=chunk_size,
+        client=mock_client_with_semantic_text_splitter,
+    )
+    segments = sts.split_text("This is a test")
+    assert len(segments) == expected_segmentation_len
+
+
+def test_split_text__on_large_chunk_size__should_merge_chunks(
+    mock_client_with_semantic_text_splitter: Mock,
+) -> None:
+    sts_no_merge = AI21SemanticTextSplitter(
+        client=mock_client_with_semantic_text_splitter
+    )
+    sts_merge = AI21SemanticTextSplitter(
+        client=mock_client_with_semantic_text_splitter,
+        chunk_size=1000,
+    )
+    segments_no_merge = sts_no_merge.split_text("This is a test")
+    segments_merge = sts_merge.split_text("This is a test")
+    assert len(segments_merge) > 0
+    assert len(segments_no_merge) > 0
+    assert len(segments_no_merge) > len(segments_merge)
+
+
+def test_split_text__on_small_chunk_size__should_not_merge_chunks(
+    mock_client_with_semantic_text_splitter: Mock,
+) -> None:
+    sts_no_merge = AI21SemanticTextSplitter(
+        client=mock_client_with_semantic_text_splitter
+    )
+    segments = sts_no_merge.split_text("This is a test")
+    assert len(segments) == 2
+    for index in range(2):
+        assert segments[index] == SEGMENTS[index].segment_text
+
+
+def test_create_documents__on_start_index__should_should_add_start_index(
+    mock_client_with_semantic_text_splitter: Mock,
+) -> None:
+    sts = AI21SemanticTextSplitter(
+        client=mock_client_with_semantic_text_splitter,
+        add_start_index=True,
+    )
+
+    response = sts.create_documents(texts=[TEXT])
+    assert len(response) > 0
+    for segment in response:
+        assert segment.page_content is not None
+        assert segment.metadata is not None
+        assert "start_index" in segment.metadata
+        assert segment.metadata["start_index"] > -1
+
+
+def test_create_documents__on_metadata(
+    mock_client_with_semantic_text_splitter: Mock,
+) -> None:
+    sts = AI21SemanticTextSplitter(client=mock_client_with_semantic_text_splitter)
+    metadatas = [{"hello": "world"}]
+    response = sts.create_documents(texts=[TEXT], metadatas=metadatas)
+    assert len(response) > 0
+    for segment in response:
+        assert segment.page_content is not None
+        assert segment.metadata is not None
+        assert len(segment.metadata) == 2
+        assert "source_type" in segment.metadata
+        assert segment.metadata["source_type"] is not None
+
+
+def test_split_text_to_documents__on_metadata_exists(
```

Comment:
Perhaps a more accurate name would be `when_metadatas_not_passed__should_contain_metadata_in_segment`?